File: src/WikiParse/src/DataModel/Citation.php
<?php

namespace WikiConnect\ParseWiki\DataModel;

class Citation
{
    private string $text;
    private string $options;
    private string $cite_text;
    public function __construct(string $text, string $options = "", string $cite_text = "")
    {
        $this->text = $text;
        $this->options = $options;
        $this->cite_text = $cite_text;
    }
    public function getOriginalText(): string
    {
        return $this->cite_text;
    }
    public function getTemplate(): string
    {
        return $this->text;
    }
    public function getContent(): string
    {
        return $this->text;
    }
    public function getAttributes(): string
    {
        return $this->options;
    }
    public function toString(): string
    {
        return "<ref " . trim($this->options) . ">" . $this->text . "</ref>";
    }
}

File: python/src/WikiParse/src/DataModel/Citation.py
"""
Citation data model

Implemented from: src/WikiParse/src/DataModel/Citation.php

Usage:
    from src.WikiParse.src.DataModel.Citation import Citation
"""


class Citation:
    """
    Represents a citation in wikitext
    
    Matches: src/WikiParse/src/DataModel/Citation.php
    """
    
    def __init__(self, text: str = "", options: str = "", cite_text: str = ""):
        """
        Initialize Citation
        
        Args:
            text: Citation content
            options: Citation attributes
            cite_text: Original citation text
        """
        self.text = text
        self.options = options
        self.cite_text = cite_text
    
    def getOriginalText(self) -> str:
        """Get original citation text"""
        return self.cite_text
    
    def getTemplate(self) -> str:
        """Get template (content)"""
        return self.text
    
    def getContent(self) -> str:
        """Get content"""
        return self.text
    
    def getAttributes(self) -> str:
        """Get attributes"""
        return self.options
    
    def toString(self) -> str:
        """Convert to string"""
        opts = self.options.strip()
        if opts:
            return f"<ref {opts}>{self.text}</ref>"
        return f"<ref>{self.text}</ref>"
    
    def __str__(self) -> str:
        """String representation"""
        return self.toString()
